docs/deprecation.md

```diff
@@ -0,0 +1,23 @@
+# Deprecation Policy
```

Comment:
It feels natural to talk about breaking changes together with deprecations. I don't want to expand the scope too much. I'm just mentioning this in case you think it should affect the name of the file or the heading. I personally think it's fine to cover breaking changes without changing either.

Reply:
I've added "& Breaking Changes" to the title